sechub-cli/src/mercedes-benz.com/sechub/cli/tarbuilder.go

```diff
@@ -0,0 +1,80 @@
+// SPDX-License-Identifier: MIT
+
+package cli
+
+import (
+	"archive/tar"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	sechubUtil "mercedes-benz.com/sechub/util"
+)
+
+// createBinariesTarFile - compress all defined binaries into one single tar archive
+func createBinariesTarFile(context *Context) error {
+	tarFile, _ := filepath.Abs(context.binariesTarFileName)
+
+	// create tar file
+	newTarFile, err := os.Create(tarFile)
+	if err != nil {
+		return err
+	}
+
+	// create tar writer
+	tarWriter := tar.NewWriter(newTarFile)
+
+	// Add everything in Data.Binaries to tar file
+	for _, item := range context.sechubConfig.Data.Binaries {
+		err = appendToBinariesTarFile(tarFile, tarWriter, item, context.config.quiet, context.config.debug)
+		if err != nil {
+			return err
+		}
+	}
+
+	tarWriter.Close()
+	newTarFile.Close()
+
+	// Check if context.binariesTarFileName is an empty tar
+	// For performance reasons we only look deeper into very small files
+	if sechubUtil.GetFileSize(tarFile) < 10000 {
```

Comment:
I also thought about this, but it would be extra effort: new field in the struct, counting, handling between the `cli` and the `util` module, so I did it this way.

The file's metadata will be read from OS for the upload anyway.
Subsequent reads will be served from OS's cache.

One could count the files (in both: binaries tar file and sources zip) and display it for the user.
If you find this a useful feature, please open a new issue for this.